📖 Add CAPI book section on `clusterctl alpha rollout` (#4328)
kubernetes-sigs/cluster-api: docs/book/src/clusterctl/commands/alpha-rollout.md

# clusterctl alpha rollout

> The `clusterctl alpha rollout` command manages the rollout of a cluster-api resource. It consists of several sub-commands which are documented below. 

Reviewer: maybe we should make `cluster-api OR Cluster API`  word consistent throughout this document? (`Cluster API` seems to be more frequently used in the repo)

<aside class="note">

<h1> Valid Resource Tpes </h1>

Currently, only the following cluster API resources are supported by the rollout command:

- machinedeployment

</aside>

### Restart 

> Use the `restart` sub-command to force an immediate rollout. For example, here the MchineDeployment `my-md-0` will be immediately rolled out:

Reviewer 2: Should we add a note explaining what rollout means: deleting machines (including cordon/drain of workloads), recreating new ones according to maxUnavailable/maxSurge

> Use the `restart` sub-command to force an immediate rollout. For example, here the MchineDeployment `my-md-0` will be immediately rolled out:

Reviewer: ```suggestion
Use the `restart` sub-command to force an immediate rollout. For example, here the MachineDeployment `my-md-0` will be immediately rolled out:
```

```
clusterctl alpha rollout restart machinedeployment/my-md-0
```

### Undo

> Use the `undo` sub-command to rollback to an ealier revision. For example, here the MachineDeployment `my-md-0` will be rolled back to revison number 3. If the `--to-revision` flag is omited, the MachineDeployment will be rolled back to the revision immediately preceding the current one. 

Reviewer 2: should we document what happen if the requested revision/the immediate preceding revision does not exists?

> Use the `undo` sub-command to rollback to an ealier revision. For example, here the MachineDeployment `my-md-0` will be rolled back to revison number 3. If the `--to-revision` flag is omited, the MachineDeployment will be rolled back to the revision immediately preceding the current one. 

Reviewer: ```suggestion
Use the `undo` sub-command to rollback to an earlier revision. For example, here the MachineDeployment `my-md-0` will be rolled back to revision number 3. If the `--to-revision` flag is omitted, the MachineDeployment will be rolled back to the revision immediately preceding the current one. 
```

```

> clusterctl alpha rollout undo machinedeployment/my-md-0 --to-revision=3`

Reviewer: ```suggestion
clusterctl alpha rollout undo machinedeployment/my-md-0 --to-revision=3
```

```

### Pause/Resume

> Use the `pause` sub-command to pause a Cluster API resource. 

Reviewer 2: Should we document what happens if if are pausing an already paused rollout (or if we are unpausing something not currently paused)

Author: Done

```
clusterctl alpha rollout pause machinedeployment/my-md-0
```

> Use th `resume` sub-command to resume a currently paused Cluster API resource. 

Reviewer: ```suggestion
Use the `resume` sub-command to resume a currently paused Cluster API resource. 
```

```
clusterctl alpha rollout resume machinedeployment/my-md-0
```

<aside class="note warning">

<h1> Warning </h1>

> Paused resources will not be reconciled by a controller. By resuming a resource, we allow it to be reconciled again. 

Reviewer 2: Are we using paused at object level or machineDeployments.spec.Paused?
In case we are using the first, should we document that pausing could imply also some undesired side effects because nothing gets updated, including e.g. the ObservedGeneration, replica counters etc.
